### PR TITLE
Bug 1380389 - autosize comments should be a user pref (default: on)

### PR DESCRIPTION
**Pull request by umohm1**

### CODE REVIEW

**Comment by floatingatoll** [comment]
Isn't this already covered by the existing preference for "auto-enlarge comment area"? (Or is there no such pref.)

**Comment by dylanwh** [comment]
No, the pref is for zooming. This is a new behavior that wlach wrote: https://twitter.com/wlach/status/885172638183809024

File: extensions/BugModal/web/bug_modal.js [hunk 1]
```diff
@@ -875,6 +875,21 @@ $(function() {
     // auto-enlarge comment area (up to its max-height)
     autosize($('#comment'));
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: You need to remove these two lines, we don't want that code to run.

File: extensions/BugModal/web/bug_modal.js [hunk 2]
```diff
@@ -875,6 +875,21 @@ $(function() {
...
 
+    // calling autosize.update or autosize
+    if (BUGZILLA.user.settings.autosize_comment) {
+    autosize.update($('#comment'));
+    }
+
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: remove the above four lines, and put them up near the existing call to autosize.update($('#comment'));

File: extensions/BugModal/web/bug_modal.js [hunk 3]
```diff
@@ -875,6 +875,21 @@ $(function() {
...
+    if (BUGZILLA.user.settings.autosize_comments) {
+     autosize($('#comment'));
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: nit: fix indentation, this should be four spaces per level

File: Bugzilla/Install.pm [hunk 4]
```diff
@@ -181,6 +181,12 @@ sub SETTINGS {
         default  => 'off',
         category => 'Searching'
     },
+    {
+        name     => 'autosize_comments',
+        options  => ['on', 'off'],
+        default  => 'on',
+        category => 'Autosize Comments'
```
Comment (dylanwh) [inline Bugzilla/Install.pm]: Category should be 'User Interface'

File: extensions/BugModal/web/bug_modal.js [hunk 5]
```diff
@@ -868,12 +868,20 @@ $(function() {
...
-            $.scrollTo($('#comment'), function() { $('#comment').focus(); });
-        });
-
-    // auto-enlarge comment area (up to its max-height)
-    autosize($('#comment'));
+            
+            // calling autosize.update or autosize
+            if (BUGZILLA.user.settings.autosize_comment) {
+                autosize.update($('#comment'));
+            }
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: structure got all messed up here.
```javascript
if (BUGZILLA.user.settings.autosize_comment) {
    autosize.update($('#comment'));
}
```
needs to happen inside the un-named function/callback ```.click(function(event) {``` which begins on line 846. After that, there needs to be a }) that closes off that function.

Then the below code happens -- which will then be one indentation level lower.

File: extensions/BugModal/template/en/default/bug_modal/field.html.tmpl [hunk 6]
```diff
@@ -219,7 +219,7 @@ END;
 
         [% CASE constants.FIELD_TYPE_FREETEXT %]
           [%# normal input field %]
-          <input name="[% name FILTER html %]" id="[% name FILTER html %]" value="[% value FILTER html %]" [% aria_labelledby_html FILTER none %] [% "required" IF required %]>
+          <input name="[% name FILTER html %]" id="[% name FILTER html %]" value="[% value FILTER html %]" [% aria_labelledby_html FILTER none +%] [% " required" IF required %]>
         [% CASE constants.FIELD_TYPE_USER %]
```
Comment (dylanwh) [inline extensions/BugModal/template/en/default/bug_modal/field.html.tmpl]: This is an unrelated change, which we can be fixed by doing a 'git fetch origin' followed by a `git rebase -i origin/master` and deleting the comment named `invalid html - missing space before "required" attribute`

File: extensions/BugModal/web/bug_modal.js [hunk 7]
```diff
@@ -857,6 +857,10 @@ $(function() {
             else if (BUGZILLA.user.settings.quote_replies == 'simply_reply') {
                 reply_text = prefix;
             }
+            if (BUGZILLA.user.settings.autosize_comment) {
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: this needs to be moved down below 
```javascript
     $('#comment').val($('#comment').val() + reply_text);
}
```

File: extensions/BugModal/web/bug_modal.js [hunk 8]
```diff
@@ -868,12 +868,21 @@ $(function() {
             if ($('#comment').val() != reply_text) {
                 $('#comment').val($('#comment').val() + reply_text);
             }
-            autosize.update($('#comment'));
-            $.scrollTo($('#comment'), function() { $('#comment').focus(); });
-        });
+            if (BUGZILLA.user.settings.autosize_comment) {
+                autosize.update($('#comment'));
+            }
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: add:
```javasript
 $.scrollTo($('#comment'), function() { $('#comment').focus(); });
```

File: extensions/BugModal/template/en/default/bug_modal/field.html.tmpl [hunk 9]
```diff
@@ -219,7 +219,7 @@ END;
 
         [% CASE constants.FIELD_TYPE_FREETEXT %]
           [%# normal input field %]
-          <input name="[% name FILTER html %]" id="[% name FILTER html %]" value="[% value FILTER html %]" [% aria_labelledby_html FILTER none +%] [% " required" IF required %]>
+          <input name="[% name FILTER html %]" id="[% name FILTER html %]" value="[% value FILTER html %]" [% aria_labelledby_html FILTER none +%] [% "required" IF required %]>
```
Comment (dylanwh) [inline extensions/BugModal/template/en/default/bug_modal/field.html.tmpl]: this change snuck back in. please add the space before the "required" string, so it is " required"

File: extensions/BugModal/web/bug_modal.js [hunk 10]
```diff
@@ -868,12 +868,23 @@ $(function() {
...
+            // calling autosize.update or autosize
+            if (BUGZILLA.user.settings.autosize_comments) {
+                autosize($('#comment'));
+            }
             $.scrollTo($('#comment'), function() { $('#comment').focus(); });
-        });
 
-    // auto-enlarge comment area (up to its max-height)
-    autosize($('#comment'));
+            else if (BUGZILLA.user.settings.zoom_textareas) {
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: this else belongs after the }, and it isn't valid to have code between } and an else.

File: extensions/BugModal/web/bug_modal.js [hunk 11]
```diff
@@ -868,12 +868,22 @@ $(function() {
...
+            // calling autosize.update or autosize
+            if (BUGZILLA.user.settings.autosize_comments) {
+                autosize($('#comment'));
+            }
+            else if (BUGZILLA.user.settings.zoom_textareas) {
+                // add comment --> enlarge on focus
+                $('#comment').focus(function(event) {
+                    $(event.target).attr('rows', 25);
+                }
+            });
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: move this up

File: extensions/BugModal/web/bug_modal.js [hunk 12]
```diff
@@ -868,12 +868,23 @@ $(function() {
             if ($('#comment').val() != reply_text) {
                 $('#comment').val($('#comment').val() + reply_text);
             }
-            autosize.update($('#comment'));
+            if (BUGZILLA.user.settings.autosize_comment) {
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: typo: autosize_comment vs. autosize_comments

File: extensions/BugModal/web/bug_modal.js [hunk 13]
```diff
@@ -868,12 +868,23 @@ $(function() {
...
+                autosize.update($('#comment'));
+            }
             $.scrollTo($('#comment'), function() { $('#comment').focus(); });
         });
 
-    // auto-enlarge comment area (up to its max-height)
-    autosize($('#comment'));
+            // calling autosize.update or autosize
```
Comment (dylanwh) [inline extensions/BugModal/web/bug_modal.js]: this comment isn't useful, we can remove it.

**Comment by dylanwh** [comment]
Made a PR for some formatting changes: https://github.com/umohm1/bmo/pull/3

**Pull request merged**